recipes/sobjectizer/all/conanfile.py

```diff
@@ -65,42 +72,44 @@ def validate(self):
                 "%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
             return
 
-        version = tools.Version(self.settings.compiler.version)
+        version = Version(self.settings.compiler.version)
         if version < minimal_version[compiler]:
             raise ConanInvalidConfiguration("%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["SOBJECTIZER_BUILD_SHARED"] = self.options.shared
+        tc.variables["SOBJECTIZER_BUILD_STATIC"] = not self.options.shared
+        tc.variables["SOBJECTIZER_INSTALL"] = True
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0091"] = "NEW"
+        tc.generate()
 
-        self._cmake = CMake(self)
-        self._cmake.definitions["SOBJECTIZER_BUILD_SHARED"] = self.options.shared
-        self._cmake.definitions["SOBJECTIZER_BUILD_STATIC"] = not self.options.shared
-        self._cmake.definitions["SOBJECTIZER_INSTALL"] = True
+        cmake_deps = CMakeDeps(self)
+        cmake_deps.generate()
 
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def build(self):
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
+        cmake.configure(build_script_folder=os.path.join(self.source_folder, "dev", "so_5"))
         cmake.build()
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def package(self):
-        cmake = self._configure_cmake()
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
         cmake.install()
-        self.copy("license*", src=self._source_subfolder, dst="licenses",  ignore_case=True, keep_path=False)
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
 
     def package_info(self):
         cmake_target = "SharedLib" if self.options.shared else "StaticLib"
```

Comment:
I would add a cmake alias for sobjectizer::sobjectizer as well, so the same target can be used regardless of whether a shared or static library was built.

Reply:
It is to emulate the original targets in projects own cmake-files.

Reply:
Oh, absolutely. My suggestion was to add an alias in addition to these. Or do you think it's not a good idea due to compatibility and portability reasons, perhaps?